tools/pio/copy_files.py

```diff
@@ -25,9 +25,9 @@ def get_max_bin_size(env_name, file_suffix):
             # Factory bin files include a part which is not overwritten via OTA
             max_bin_size = max_bin_size + 65536
     if "_ESP32_" in env_name or "_ESP32s2_" in env_name or "_ESP32s3_" in env_name:
-        if "_8M2M" in env_name:
-            # ESP32 with 2944k of sketch space.
-            max_bin_size = 3080192
+        if "_8M1M" in env_name:
+            # ESP32 with 3520k of sketch space.
+            max_bin_size = 3604480
```

Comment:
This is the factory bin size, right?
3604480 is the actual size of 0x370000
So with this check it might in theory accept the non-factory bin file and not the factory file. But flashing such a falsely accepted build may still cause issues.

Reply:
We might need to add the `factory` part when checking the factory.bin files, as these numbers are the size of the `app` partitions.